nengo/tests/test_params.py

```diff
@@ -322,3 +323,63 @@ class Test(object):
         obsolete = params.ObsoleteParam('obsolete', 'not included in params')
 
     assert set(params.iter_params(Test())) == {'p1', 'p2'}
+
+
+def test_coerce_defaults():  # noqa: C901
```

Comment:
I think we might potentially be able to come up with a better name for this test. As far as I can tell it is not really testing the correct coercion of defaults, but rather that setting values in a config does not fail if the coercion cannot be done at that point (because missing information about parent objects).

Reply:
No strong opinions, feel free to change it to whatever you'd like.